### PR TITLE
docs(meshmetrics): initial MeshMetric docs

### PR DESCRIPTION
**Pull request by slonka**
Closes https://github.com/kumahq/kuma-website/issues/1577

Policy doc: https://deploy-preview-1578--kuma.netlify.app/docs/dev/policies/meshmetric/

Introduction: https://deploy-preview-1578--kuma.netlify.app/docs/dev/policies/introduction/ 

Observability: https://deploy-preview-1578--kuma.netlify.app/docs/dev/explore/observability/

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits) yes

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? yes

### CODE REVIEW

File: app/_src/policies/meshmetric.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+{% warning %}
+To use `clientId` setting you need to be running at least Prometheus `2.49.0`.
+{% endwarning %}
+
+Example configurations differentiated by `prometheus` tag:
+
+{% policy_yaml first %}
+```yaml
+type: MeshMetrics
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:179]:
```suggestion
type: MeshMetric
```

File: app/_src/policies/meshmetric.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        prometheus: 
+          clientId: "prometheus-one" 
+          port: 5670
+          path: /metrics
+```
+{% endpolicy_yaml %}
+
+{% policy_yaml second %}
+```yaml
+type: MeshMetrics
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:198]:
```suggestion
type: MeshMetric
```

File: app/_src/policies/meshmetric.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+In addition to the `MeshMetric` configuration, `kuma-sidecar` requires a provided certificate and key for its operation.
+
+{% tabs expose-mesh-metrics-data-plane-proxies-tls useUrlFragment=false %}
+{% tab expose-mesh-metrics-data-plane-proxies-tls Kubernetes %}
+
+When the certificate and key are available within the container, `kuma-sidecar` needs the paths to provided files as the following environment variables:
+
+* KUMA_DATAPLANE_RUNTIME_METRICS_CERT_PATH
+* KUMA_DATAPLANE_RUNTIME_METRICS_KEY_PATH
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:126]:
```suggestion
* `KUMA_DATAPLANE_RUNTIME_METRICS_CERT_PATH`
* `KUMA_DATAPLANE_RUNTIME_METRICS_KEY_PATH`
```

File: app/_src/policies/meshmetric.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+default:
+  applications:
+    - path: "/metrics/prometheus"
+      port: 8888
+```
+{% endpolicy_yaml %}
+
+## All policy options
+
+{% json_schema MeshMetrics %}
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:284]: Shouldn't it be `MeshMetric`?

```suggestion
{% json_schema MeshMetric %}
```

File: app/_src/policies/meshmetric.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+{{site.mesh_product_name}} allows scraping Prometheus metrics from the applications endpoint running in the same `Pod` or `VM`.
+Later those metrics are aggregated and exposed at the same `port/path` as data plane proxy metrics.
+It is possible to configure it at the `Mesh` level, for all the applications in the `Mesh`, or just for specific applications.
+
+Here are reasons where you'd want to use this feature:
+- Application metrics are labelled with your mesh parameters (tags, mesh, data plane name...), this means that in mixed Universal and Kubernetes mode metrics are reported with the same types of labels.
+- Both application and sidecar metrics are scraped at the same time. This makes sure they are coherent (with 2 different scrapers they can end up scraping at different intervals and make metrics harder to correlate).
+- If you disable [passthrough](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing) and your mesh uses mTLS and Prometheus is outside the mesh this is the only way to retrieve these metrics as the app is completely hidden behind the sidecar.
+
+[//]: # (make sure the sentence above is correct since it's no longer possible to )
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:77]:
```suggestion
```

File: app/_src/policies/meshmetric.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        }'
+```
+
+{% endtab %}
+{% tab expose-mesh-metrics-data-plane-proxies-tls Universal %}
+
+Please upload the certificate and the key to the machine, and then define the following environment variables with the correct paths:
+
+	* KUMA_DATAPLANE_RUNTIME_METRICS_CERT_PATH
+	* KUMA_DATAPLANE_RUNTIME_METRICS_KEY_PATH
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:158]:
```suggestion
	* `KUMA_DATAPLANE_RUNTIME_METRICS_CERT_PATH`
	* `KUMA_DATAPLANE_RUNTIME_METRICS_KEY_PATH`
```

File: app/_src/policies/meshmetric.md [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+{% endtab %}
+{% endtabs %}
+
+#### activeMTLSBackend
+
+We no longer support activeMTLSBackend, if you need to encrypt and authorize the metrics use [Secure metrics with TLS](#secure-metrics-with-tls) with a combination of [one of the authorization methods](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+
+#### Running multiple prometheus instances
+
+If you need to run multiple instances of Prometheus and want to target different set of DPPs you can do this by using Client ID setting on both `MeshMetric` (`clientId`) and [Prometheus configuration](https://github.com/prometheus/prometheus/pull/13278/files#diff-17f1012e0c2fbd9bcd8dff3c23b18ff4b6676eef3beca6f8a3e72e6a36633334R2233) (`client_id`).
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:169]: I think we should use the full names whenever the abbreviation is not generally well known

```suggestion
If you need to run multiple instances of Prometheus and want to target different set of Data Plane Proxies you can do this by using Client ID setting on both `MeshMetric` (`clientId`) and [Prometheus configuration](https://github.com/prometheus/prometheus/pull/13278/files#diff-17f1012e0c2fbd9bcd8dff3c23b18ff4b6676eef3beca6f8a3e72e6a36633334R2233) (`client_id`).
```

File: app/_src/policies/meshmetric.md [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+There are three main sections of the configuration: `sidecar`, `applications`, `backends`.
+The first two define how to scrape parts of the mesh (sidecar and underlying applications), the third one defines what to do with the data (in case of Prometheus instructs to scrape specific address, in case of OpenTelemetry defines where to push data).
+
+{% tip %}
+In contrast to [Traffic Metrics](/docs/{{ page.version }}/policies/traffic-metrics) all configuration is dynamic and no restarts of the DPPs are needed.
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:45]: I think we should use the full names whenever the abbreviation is not generally well known
```suggestion
In contrast to [Traffic Metrics](/docs/{{ page.version }}/policies/traffic-metrics) all configuration is dynamic and no restarts of the Data Plane Proxies are needed.
```

File: app/_src/policies/meshmetric.md [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+        client_id: "prometheus-two"
+```
+
+## Examples
+
+### With custom port, path, clientId, application aggregation and service override
+
+The first policy defines a default `MeshMetric` policy for the `default` mesh.
+The second policy creates an override for workloads tagged with `framework: example-web-framework`.
+That web framework exposes metrics under `/metrics/prometheus` and port 8888.
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:243]:
```suggestion
That web framework exposes metrics under `/metrics/prometheus` and port `8888`.
```

File: app/_src/policies/meshmetric.md [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+You can define metrics configuration for a whole Mesh, and optionally tweak certain parts for individual data plane proxies.
+For example, you might need to override the default metrics port if it's already in use on the specified machine.
+
+{{site.mesh_product_name}} provides full integration with Prometheus:
+
+* Each proxy can expose its metrics in [Prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).
+* {{site.mesh_product_name}} exposes an API called the monitoring assignment service (MADS) which exposes proxies configured by `MeshMetric`.
+
+To collect metrics from {{site.mesh_product_name}}, you need to expose metrics from proxies and applications.
+
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:21]:
```suggestion
```

**Review by bartsmykla — CHANGES_REQUESTED** [review]
Few nits, but `type: MeshMetrics` needs to be fixed

File: app/_src/policies/meshmetric.md [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+- If you disable [passthrough](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing) and your mesh uses mTLS and Prometheus is outside the mesh this is the only way to retrieve these metrics as the app is completely hidden behind the sidecar.
+
+[//]: # (make sure the sentence above is correct since it's no longer possible to )
+
+Example section of the configuration:
+
+```yaml
+    applications:
+      - path: "/metrics/prometheus"
+        address: # optional custom address if the underlying application listens on a different address than the DPP
```
Comment (bartsmykla) [inline app/_src/policies/meshmetric.md:84]:
```suggestion
        address: # optional custom address if the underlying application listens on a different address than the Data Plane Proxy
```

File: app/_src/policies/meshmetric.md (hunk 4 above)
Comment (slonka) [reply app/_src/policies/meshmetric.md:284]: The schema rendered - https://deploy-preview-1578--kuma.netlify.app/docs/dev/policies/meshmetric/#all-policy-options so I don't think so, this one needs to be plural.

Comment (bartsmykla) [reply app/_src/policies/meshmetric.md:281]: Makes sense!

**Review by bartsmykla — APPROVED** [review]
lgtm now

**Pull request merged**